internal/services/helpers.go

```diff
@@ -134,3 +141,102 @@ func ReturnErr(w http.ResponseWriter, err string, code int) {
 	w.WriteHeader(code)
 	json.NewEncoder(w).Encode(errj)
 }
+func HashedPassword(password string) (error, []byte) {
+	hashedPassword, err := bcrypt.GenerateFromPassword([]byte(password), bcrypt.DefaultCost)
+	if err != nil {
+		return err, nil
+	}
+	return nil, hashedPassword
+}
+
+func HasDigit(password string) bool {
+	for _, char := range password {
+		if unicode.IsDigit(char) {
+			return true
+		}
+	}
+	return false
+}
+
+func HasAllowedSpecialChar(password string) bool {
+	allowedSpecialChars := []rune{'!', '@', '#', '$'}
+	for _, char := range password {
+		if unicode.IsPunct(char) && containsRune(allowedSpecialChars, char) {
```

Comment:
that's most likely wrong, or not doing what is expected. What's the point of 2 checks here?